pkg/controller/importconfig/agentregistration_manifests.go

```diff
@@ -0,0 +1,85 @@
+package importconfig
+
+import (
+	"context"
+	"encoding/base64"
+	"os"
+
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
+	corev1 "k8s.io/api/core/v1"
+	operatorv1 "open-cluster-management.io/api/operator/v1"
+)
+
+func GenerateKlusterletCRDsV1() []byte {
+	return filesToBytes(klusterletCrdsV1File)
+}
+
+func GenerateKlusterletCRDsV1Beta1() []byte {
+	return filesToBytes(klusterletCrdsV1beta1File)
+}
+
+func GenerateAgentRegistrationManifests(ctx context.Context, clientHolder *helpers.ClientHolder, clusterID string) ([]byte, error) {
```

Comment:
add a comment here. this function is only for registration trigger by agent I think? why it cannot be a common func?

Reply:
This function is likely a construct function for agent-registration case. 

Compare with another functions `generateImportSecret`(this one also render files but finally store them as a secret.) and `func (w *hostedWorker) generateImportSecret`.

If we consider three of them, actually they are sharing very few common functions. 

We can create a common function for "agent-registration", "default", "hosted", but that's also means we will add depth and complexity for code reading. So I kindly feel we can wait util we are in "deprecateing image pull secret" part or as least "KlusterletConfig" phase to sense more common parts of the three.

Reply:
In the final version, we will have a collection of common functions which is `importconfig`. The `agent-registration` and `hosted` specific logic will go to their repo.